### PR TITLE
fix: arrow issues and some improvements

### PR DESCRIPTION
**Pull request by Manukyanq**
This PR contains fixes and improvements related to arrows over the board

- [x]  fixes  https://github.com/Clariity/react-chessboard/issues/94 issue.
- [x] allows set color for each arrow like was asked in https://github.com/Clariity/react-chessboard/issues/74 issue
- [x] completes fixes from https://github.com/Clariity/react-chessboard/pull/81 pull request 

https://github.com/Clariity/react-chessboard/assets/42891870/97fd7278-ef7f-40a2-9877-99daac6d5baa

### CODE REVIEW

File: src/chessboard/components/Arrows.tsx [hunk 1]
```diff
@@ -28,16 +28,26 @@ export const Arrows = () => {
...
+        );
+        const to = getRelativeCoords(
+          boardOrientation,
+          boardWidth,
+          arrowEndField
+        );
         let ARROW_LENGTH_REDUCER = boardWidth / 32;
 
         const isArrowActive = i === arrows.length;
         // if there are differnet arrows targeting same square make thier length a bit shorter
```
Comment (Clariity) [inline src/chessboard/components/Arrows.tsx:46]:
```suggestion
        // if there are different arrows targeting the same square make their length a bit shorter
```

File: src/chessboard/hooks/useArrows.ts [hunk 2]
```diff
@@ -1,86 +1,82 @@
...
-  onArrowsChange?: (arrows: Arrows) => void
+  onArrowsChange?: (arrows: Arrows) => void,
+  customArrowColor?: string
 ) => {
-  // current arrows
-  const [arrows, setArrows] = useState(new Set<string>());
+  // arrows passed programatically to `ChessBoard` as a react prop
+  const [customArrowsSet, setCustomArrows] = useState<Arrows>([]);
+
+  // arrows drawed with mouse by user on the board
```
Comment (Clariity) [inline src/chessboard/hooks/useArrows.ts:15]:
```suggestion
  // arrows drawn with mouse by user on the board
```

File: src/chessboard/hooks/useArrows.ts [hunk 3]
```diff
@@ -1,86 +1,82 @@
...
+
+    // add the newArrow to arrows array if it is unique
+    if (isNewArrowUnique) {
+      arrowsCopy = [...arrows, newArrow];
+    }
+    // remove it from the board if we already have same arrow in arrows array
     else {
-      arrowsSet.add(newArrow);
+      arrowsCopy = arrows.filter(([arrowFrom, arrowTo]) => {
+        debugger;
```
Comment (Clariity) [inline src/chessboard/hooks/useArrows.ts:67]: this can be removed

File: src/chessboard/types/index.ts [hunk 4]
```diff
@@ -165,10 +167,11 @@ export type ChessboardProps = {
    */
   clearPremovesOnRightClick?: boolean;
   /**
-   * Array of custom arrows to draw on the board. Each arrow within the array must be an array of length 2 with strings denoting the from and to square to draw the arrow e.g. [ ['a3', 'a5'], ['g1', 'f3'] ].
+   * Array of custom arrows to draw on the board. Each arrow within the array must be an array of length 2 with `Square` denoting the from and to square to draw the arrow e.g. [ ['a3', 'a5'], ['g1', 'f3', 'red'] ].
```
Comment (Clariity) [inline src/chessboard/types/index.ts:170]: please update as this is no longer accurate, each array doesn't have to be of length 2 and Square is no longer used for typing

**Comment by Clariity** [comment]
awesome stuff, please update the readme to reflect changes

File: src/chessboard/hooks/useArrows.ts [hunk 5]
```diff
@@ -1,86 +1,81 @@
...
     if (fromSquare === toSquare) return;
-    // remove it if we already have same arrow in arrows set
-    const newArrow = `${fromSquare},${toSquare}`;
-    const arrowsSet = new Set(arrows);
-    if (arrowsSet.has(newArrow)) {
-      arrowsSet.delete(newArrow);
-    } // add to arrows set  new arrow
+
+    let arrowsCopy;
+    const newArrow = [fromSquare, toSquare, customArrowColor] as Arrow;
```
Comment (Clariity) [inline src/chessboard/hooks/useArrows.ts:54]:
```suggestion
    const newArrow: Arrow = [fromSquare, toSquare, customArrowColor];
```

does this work?

Comment (Manukyanq) [reply src/chessboard/hooks/useArrows.ts:54]: yes

**Review by Clariity — APPROVED** [review]

**Pull request merged**